Simulator/Layouts/Layout.h

```diff
@@ -67,28 +67,28 @@ class Layout {
    /// @return The number of vertices managed by the Layout
    virtual int getNumVertices() const;
 
-   VectorMatrix *xloc_;   ///< Store neuron i's x location.
+   VectorMatrix xloc_;   ///< Store neuron i's x location.
 
-   VectorMatrix *yloc_;   ///< Store neuron i's y location.
+   VectorMatrix yloc_;   ///< Store neuron i's y location.
 
-   CompleteMatrix *dist2_;   ///< Inter-neuron distance squared.
+   CompleteMatrix dist2_;   ///< Inter-neuron distance squared.
 
-   CompleteMatrix *dist_;   ///< The true inter-neuron distance.
+   CompleteMatrix dist_;   ///< The true inter-neuron distance.
 
    vector<int>
       probedNeuronList_;   ///< Probed neurons list. // ToDo: Move this to Hdf5 recorder once its implemented in project -chris
 
-   vertexType *vertexTypeMap_;   ///< The vertex type map (INH, EXC).
+   vector<vertexType> vertexTypeMap_;   ///< The vertex type map (INH, EXC).
 
-   bool *starterMap_;   ///< The starter existence map (T/F).
+   vector<bool> starterMap_;   ///< The starter existence map (T/F).
```

Comment:
Will this cause problems with GPU implementation, or is just used during initialization (implemented using CPU code)?

Reply:
It did cross my mind, so I did check if starterMap_ was being used in GPU code and it wasn't. Just to be on the safe side do you prefer this to be a unique_ptr to bool[ ] ?

Reply:
No, this is fine.